SPEC.md

```diff
@@ -334,32 +365,236 @@ interpreted as described in the following table:
+desired.
+
+A Multi-line string _MUST_ start with a [Newline](#newline) immediately
+following its opening `"`. Its final line _MUST_ contain only whitespace,
+followed by a single closing `"`. All in-between lines that contain
+non-newline characters _MUST_ start with the exact same whitespace as the
+final line (precisely matching codepoints, not merely counting characters).
+
+The value of the Multi-line String omits the first and last Newline, the
+Whitespace of the last line, and the matching Whitespace prefix on all
+intermediate lines. The first and last Newline can be the same character (that
+is, empty multi-line strings are legal).
+
+Strings with literal Newlines that do not immediately start with a Newline and
+whose final `"` is not preceeded by optional whitespace and a Newline are
+illegal.
+
+In other words, the final line specifies the whitespace prefix that will be
+removed from all other lines.
+
+It is a syntax error for any body lines of the multi-line string to not match
+the whitespace prefix of the last line with the final quote.
+
+#### Newline Normalization
+
+Literal Newline sequences in Multi-line Strings must be normalized to a single
+`U+000A` (`LF`) during deserialization. This means, for example, that `CR LF`
+becomes a single `LF` during parsing.
+
+This normalization does not apply to non-literal Newlines entered using escape
+sequences.
 
-Raw String literals are represented as `r`, followed by zero or more `#`
-characters, followed by `"`, followed by any number of UTF-8 literals. The
-string is then closed by a `"` followed by a _matching_ number of `#`
-characters. This allows them to contain raw `"` or `#` characters; only the
-precise terminator (resembling `"##`, for example) ends the raw string. This
-means that the string sequence `"` or `"#` and such must not match the closing
-`"` with the same or more `#` characters as the opening `r`.
+For clarity: this normalization is for individual sequences. That is, the
+literal sequence `CRLF CRLF` becomes `LF LF`, not `LF`.
 
 #### Example
 
 ```kdl
-just-escapes r"\n will be literal"
-quotes-and-escapes r#"hello\n\r\asd"world"#
+multi-line "
+        foo
+    This is the base indentation
+            bar
+    "
+```
+
+This example's string value will be:
+
+```
+    foo
+This is the base indentation
+        bar
+```
+
+which is equivalent to `"    foo\nThis is the base indentation\n        bar"`
+when written as a single-line string.
+
+---------
+
+If the last line wasn't indented as far,
+it won't dedent the rest of the lines as much:
+
+```kdl
+multi-line "
+        foo
+    This is no longer on the left edge
+            bar
+  "
+```
+
+This example's string value will be:
+
+```
+      foo
+  This is no longer on the left edge
+          bar
+```
+
+Equivalent to `"      foo\n  This is no longer on the left edge\n          bar"`.
+
+-----------
+
+Empty lines can contain any whitespace, or none at all, and will be reflected as empty in the value:
```

Comment:
I don't think this matches the description above or the discussion at https://github.com/kdl-org/kdl/pull/286/files#r1483699764

Should this say something like "Empty lines may omit the whitespace prefix:" instead?